src/main/java/fi/aalto/cs/apluscourses/intellij/model/CourseProject.java

```diff
@@ -0,0 +1,44 @@
+package fi.aalto.cs.apluscourses.intellij.model;
+
+import com.intellij.openapi.project.Project;
+import fi.aalto.cs.apluscourses.intellij.notifications.DefaultNotifier;
+import fi.aalto.cs.apluscourses.intellij.services.PluginSettings;
+import fi.aalto.cs.apluscourses.model.Course;
+import fi.aalto.cs.apluscourses.utils.Event;
+import java.net.URL;
+import org.jetbrains.annotations.NotNull;
+
+public class CourseProject {
```

Comment:
I haven't worked with the code for a while, so got a bit confused here... Initially I was like "wow, finally a `CourseProject` class", but then I started to think what is its interrelation with others? 😉 

(continues after the pics)
I've made some old-fashioned pen-and-paper sketches, to support the discussion:
![IMG_20210225_091220](https://user-images.githubusercontent.com/9991098/109117694-ffedec80-774a-11eb-8aa1-e1c1dd573f63.jpg)
![IMG_20210225_091427](https://user-images.githubusercontent.com/9991098/109117696-011f1980-774b-11eb-8a7e-49e5edf782be.jpg)

As `IntelliJCourse` is a `Course`, can I claim that `CourseProject` has a `IntelliJCourse` 🤣 ? In other words, is there a way to perhaps simplify and flatten the structure or maybe introduce more distinctive names and `JavaDoc`s for classes? For example this is `AplusProject` class, it is a representation of ..., use it for .., well, actually, this is a good example, why should it be used (and what does it name tells in the context of the plugin)?

Reply:
additionally, what is the relation between `CourseProject` and a `Project`, should there be some?

Reply:
Nice drawing :smile: And the understanding from the drawing is completely correct.

I now realize that the name `CourseProject` is pretty confusing. As you mentioned, it doesn't even contain a reference to a project, but just passes it over to course updater.

What it really is, is a combination of a course and an updater that regularly updates the course. I'm not sure yet what a good name would be :thinking: 

I'll at least add JavaDocs for the classes, because as you mention, it's getting a bit confusing :smile: